consensus/consortium/v2/consortium_test.go

```diff
@@ -2269,3 +2270,53 @@ func TestIsTrippEffective(t *testing.T) {
 		t.Error("fail test Tripp effective")
 	}
 }
+
+func TestHeaderExtraDataCheckAfterTripp(t *testing.T) {
+	c := Consortium{
+		chainConfig: &params.ChainConfig{
+			TrippBlock: common.Big0,
+		},
+		config: &params.ConsortiumConfig{
+			EpochV2: 200,
+		},
+		isTest:             true,
+		testTrippEffective: true,
+	}
+
+	// Not an epoch block, every validator field must be empty
+	header := types.Header{Number: big.NewInt(100)}
+	extraData := finality.HeaderExtraData{
+		CheckpointValidators: []finality.ValidatorWithBlsPub{
+			{},
+		},
+	}
+	err := c.verifyValidatorFieldsInExtraData(nil, &extraData, &header)
+	if err == nil {
+		t.Fatalf("Expect an error")
+	}
+
+	extraData = finality.HeaderExtraData{
+		BlockProducers: []common.Address{
+			{},
+		},
+	}
+	err = c.verifyValidatorFieldsInExtraData(nil, &extraData, &header)
+	if err == nil {
+		t.Fatalf("Expect an error")
+	}
+
+	// Not a period block, checkpoint validators must be empty
+	header = types.Header{Number: big.NewInt(200)}
+	extraData = finality.HeaderExtraData{
+		CheckpointValidators: []finality.ValidatorWithBlsPub{
+			{},
+		},
+		BlockProducers: []common.Address{
+			{},
+		},
+	}
+	err = c.verifyValidatorFieldsInExtraData(nil, &extraData, &header)
+	if err == nil {
+		t.Fatalf("Expect an error")
+	}
```

Comment:
I think we should add test for the period block as well. I don't see any other test cover the case.

Reply:
This test focuses on must be empty cases, the must have check (e.g. at period block, len(checkpointValidators) must not be 0) is redundant I think, as there are checks at Finalize. We may clean up these must have checks after this release.